Source/DafnyCore/Backends/Dafny/BuilderSyntaxTree.cs

```diff
@@ -4,27 +4,30 @@ namespace Microsoft.Dafny.Compilers {
 
   class BuilderSyntaxTree<T> : ConcreteSyntaxTree {
     public readonly T Builder;
+    public readonly DafnyCodeGenerator Compiler;
 
-    public BuilderSyntaxTree(T builder) {
+    public BuilderSyntaxTree(T builder, DafnyCodeGenerator compiler) {
       if (builder == null) {
         throw new ArgumentNullException(nameof(builder));
       }
 
       Builder = builder;
+      Compiler = compiler;
     }
 
     public override ConcreteSyntaxTree Fork(int relativeIndent = 0) {
       if (Builder is StatementContainer statementContainer) {
-        return new BuilderSyntaxTree<StatementContainer>(statementContainer.Fork());
+        return new BuilderSyntaxTree<StatementContainer>(statementContainer.Fork(), Compiler);
       } else {
-        DafnyCodeGenerator.throwGeneralUnsupported(); // Warning: this is an invalid operation: cannot fork builder of type Builder.GetType()
+        Compiler.throwGeneralUnsupported("<i>Builder not StatementContainer</i> but " + Builder.GetType().ToString());
```

Comment:
Shouldn't this be a flat out internal error instead? AFAICT it means a logic error in the implementation, not a piece of the user program that isn't yet supported.

Can be fixed in the next PR since this was already here.

Reply:
Yeah let's do that in the next PR. Thanks.